orchagent/notifications.cpp

```diff
@@ -17,15 +17,16 @@ void on_port_state_change(uint32_t count, sai_port_oper_status_notification_t *d
 
     lock_guard<mutex> lock(gDbMutex);
 
-    /* Wait until gPortsOrch is initialized */
-    if (!gPortsOrch || !gPortsOrch->isInitDone())
+    if (!gPortsOrch)
```

Comment:
then, it should be assert since it cannot happen according to what you describe.